app/workers/analyze_github_metrics_status.rb

```diff
@@ -2,45 +2,49 @@ class AnalyzeGithubMetricsStatus
   include Sidekiq::Worker
   include Rails.application.routes.url_helpers
   sidekiq_options retry: false
-  attr_reader :pull_request, :github_service
+  attr_reader :pull_request, :github_service, :project, :branch
 
   def perform(pull_request_data)
     @pull_request = GithubPullRequest.new(pull_request_data)
     handle_pull_request_status
+    return unless project.present? && branch.present?
+    GithubIssueCommenter.new.perform(project.id, branch.id, pull_request_data)
   end
 
   def handle_pull_request_status
-    project = Project.find_by(github_repo: pull_request.full_name)
-    return unless project.present?
-    fetch_project_branches(project)
-    branch = project.branches.find_by(name: pull_request.branch)
-    return unless branch.present?
-    change_pull_request_status(project, pull_request_status(branch), branch)
+    fetch_project
+    fetch_branch if project.present?
+    change_pull_request_status(pull_request_status) if branch.present?
+  end
+
+  def fetch_project
+    @project = Project.find_by(github_repo: pull_request.full_name)
+    fetch_project_branches if project.present?
   end
 
-  def change_pull_request_status(project, status, branch)
-    github_service = GithubService.new(project.admin_user)
+  def fetch_branch
+    @branch = project.branches.find_by(name: pull_request.branch)
+  end
+
+  def change_pull_request_status(status)
+    return unless branch.present?
     github_service.create_status(
       pull_request, status[:key],
       context: 'CodeStats',
-      target_url: target_url(project, branch),
+      target_url: branch.target_url,
```

Comment:
Why don't we include the url herlpers here instead of including them in the model?